src/main/java/io/temporal/samples/hello/HelloSearchAttributes.java

```diff
@@ -85,62 +116,106 @@ public String getGreeting(String name) {
     }
   }
 
+  /**
+   * Implementation of our workflow activity interface. It overwrites our defined composeGreeting
+   * activity method.
+   */
   static class GreetingActivitiesImpl implements GreetingActivities {
     @Override
     public String composeGreeting(String greeting, String name) {
       return greeting + " " + name + "!";
     }
   }
 
+  /**
+   * With our Workflow and Activities defined, we can now start execution. The main method is our
+   * workflow starter.
+   */
   public static void main(String[] args) {
-    // gRPC stubs wrapper that talks to the local docker instance of temporal service.
+    /*
+     * Define the workflow service. It is a gRPC stubs wrapper which talks to the docker instance of
+     * our locally running Temporal service.
+     */
```

Comment:
I dunno that you need to mention "the docker instance" here. In principle it could be a full on deployment, but really the fact that it talks to the Temporal service is what matters.

Also you can probably drop "Define the" from all these comments, instead just beginning with "A gRPC stubs..."